include/circt/Dialect/Calyx/CalyxPrimitives.td

```diff
@@ -513,4 +513,45 @@ def WireLibOp : UnaryLibraryOp<"wire", [SameTypeConstraint<"in", "out">]> {
   ];
 }
 
-def ExtSILibOp  : UnaryLibraryOp<"signext"> {}
+def ExtSILibOp  : UnaryLibraryOp<"signext", "std_"> {}
+
+class UnaryFloatingPointLibraryOp<string mnemonic, list<Trait> traits = []> :
+  UnaryLibraryOp<mnemonic, "", !listconcat(traits, [
+    DeclareOpInterfaceMethods<FloatingPointOpInterface>
+  ])> {}
+
+def FpToIntOpIEEE754 : UnaryFloatingPointLibraryOp<"ieee754.fptoint"> {
```

Comment:
Does this work only for 32-bit floating points, or any ieee754 format, i.e., f16, f32, f64, f128, and f256?

Reply:
yes for everything you listed except f256: https://github.com/llvm/circt/blob/2ba6439ae3e4081978b289ceffc4a78f1abeee44/lib/Dialect/Calyx/Export/CalyxEmitter.cpp#L1034 - at least from IR's perspective

There will still be some exceptions at runtime, and we can refer to section 7 of: https://www.jhauser.us/arithmetic/HardFloat-1/doc/HardFloat-Verilog.html

Reply:
yes for everything you listed except f256: https://github.com/llvm/circt/blob/2ba6439ae3e4081978b289ceffc4a78f1abeee44/lib/Dialect/Calyx/Export/CalyxEmitter.cpp#L1034 - at least from IR's perspective

There will still be some exceptions at runtime, and we can refer to section 7 of: https://www.jhauser.us/arithmetic/HardFloat-1/doc/HardFloat-Verilog.html

Reply:
Got it!